src/Cli/dotnet/Commands/Project/Convert/ProjectConvertCommand.cs

```diff
@@ -118,4 +128,85 @@ public override int Execute()
             }
         }
     }
+
+    private string DetermineOutputDirectory(string file)
+    {
+        string defaultValue = Path.ChangeExtension(file, null);
+        string defaultValueRelative = Path.GetRelativePath(relativeTo: Environment.CurrentDirectory, defaultValue);
+        string targetDirectory = _outputDirectory
+            ?? TryAskForOutputDirectory(defaultValueRelative)
+            ?? defaultValue;
+        if (Directory.Exists(targetDirectory))
+        {
+            throw new GracefulException(CliCommandStrings.DirectoryAlreadyExists, targetDirectory);
+        }
+
+        return targetDirectory;
+    }
+
+    private string? TryAskForOutputDirectory(string defaultValueRelative)
+    {
+        return InteractiveConsole.Ask<string?>(
+            string.Format(CliCommandStrings.ProjectConvertAskForOutputDirectory, defaultValueRelative),
+            _parseResult,
+            (path, out result, [NotNullWhen(returnValue: false)] out error) =>
+            {
+                if (Directory.Exists(path))
```

Comment:
Is this redundant with the check we do on 139? Or does a different, useful behavior result from also doing the check here?

Reply:
I think this causes us to prompt the user for a different directory instead of failing the entire command

Reply:
Exactly, this causes us to prompt for another directory. The check on line 139 causes us to fail in case the user provided the directory via command-line args rather than interactively.